textdb/textdb-api/src/main/java/edu/uci/ics/textdb/api/common/ITuple.java

```diff
@@ -5,8 +5,16 @@
 /**
  * Created by chenli on 3/25/16.
  */
-public interface ITuple {
+public interface ITuple extends Cloneable {
     IField getField(int index);
+
     IField getField(String fieldName);
+
     List<IField> getFields();
+
+    List<Attribute> getSchema();
+
+    void addField(Attribute attribute, IField field);

```

Comment:
Yes, I have removed the addField method from Ituple interface. Now I'm creating a new spanTuple in the DictionaryMatcher.java by modifying the schema and fields locally. Let me know if this is fine.